### PR TITLE
[READY] ports plumbing (no more waiting for tgui4)

### PR DESCRIPTION
**Pull request by lolman360**
since we have tgui4 this is all ready

https://github.com/tgstation/tgstation/pull/44833

also: automatic hydro trays. take like 20x more gaia to transcend, accept reagents via pipe, autoharvest

## Changelog
:cl: Time-Green (copypasta'd by lolman360)
add: plumbing
add: automatic hydro trays
/:cl:

### CODE REVIEW

**Comment by Tupinambis** [comment]
29 commits is a mood. Also nice PR, finally some good fucking features.

**Comment by lolman360** [comment]
you don't commit every change?

File: code/modules/research/designs/medical_designs.dm [hunk 1]
```diff
@@ -1001,7 +1001,7 @@
 	desc = "Using standard mass-energy dynamic autoconverters, generates reagents from power and puts them in a pipe."
 	id = "plumb_synth"
 	build_type = PROTOLATHE
-	materials = list(MAT_METAL = 5000, MAT_GLASS = 1000, MAT_PLASTIC = 1000)
+	materials = list(/datum/material/iron = 5000, /datum/material/glass = 1000, MAT_PLASTIC = 1000)
```
Comment (Hatterhat) [inline code/modules/research/designs/medical_designs.dm]: >MAT_PLASTIC

File: code/modules/research/designs/medical_designs.dm [hunk 2]
```diff
@@ -1100,7 +1100,7 @@
 	desc = "A tank for storing plumbed reagents."
 	id = "plumb_tank"
 	build_type = PROTOLATHE
-	materials = list(MAT_METAL = 10000, MAT_GLASS = 10000, MAT_PLASTIC = 4000)
+	materials = list(/datum/material/iron = 10000, /datum/material/glass = 10000, MAT_PLASTIC = 4000)
```
Comment (Hatterhat) [inline code/modules/research/designs/medical_designs.dm]: >MAT_PLASTIC

**Review by Hatterhat — CHANGES_REQUESTED** [review]
you left some MAT_PLASTIC defines in idiot

**Pull request merged**